tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java

```diff
@@ -322,6 +324,20 @@ private void invalidOffset(String offset) {
                 "'earliest', 'latest', or a non-negative long.");
     }
 
+    private long parseTimeoutMs() {
+        long timeout;
+        if (options.has(timeoutMsOpt)) {
+            timeout = options.valueOf(timeoutMsOpt);
+            if (timeout < 0) {
+                CommandLineUtils.printUsageAndExit(parser, "The provided timeout-ms value '" + timeout +
```

Comment:
It seems previously we wouldn't fail if a negative value was provided.

Reply:
Returned to the previous logic.